### PR TITLE
Led lights -> master

### PR DESCRIPTION
**Pull request by NoamRosenberg08**

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/Robot.java [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
-		Limelight.getInstance();
-		initPortForwarding();
-		LiveWindow.disableAllTelemetry();
-		Battery.getInstance().setDefaultCommand(new BatteryDisabler());
-		
-		//swerve
-		
-		SwerveChassis.getInstance().resetChassisPose();
-		SwerveChassis.getInstance().resetAllEncoders();
+//		Limelight.getInstance();
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Robot.java:22]: Commented out code.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/DefaultColor.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+
+import java.awt.*;
+
+public class DefaultColor extends LEDCommand {
+
+
+    public int cnt;
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/DefaultColor.java:12]: Bad naming.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/DefaultColor.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+    public DefaultColor() {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/DefaultColor.java:15]: IIRC Java creates an empty constructor automatically and you don't need to declare it yourself.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package edu.greenblitz.tobyDetermined.commands.LED;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+
+
+public class ObjectInClawLED extends LEDCommand{
+
+    public ObjectInClawLED(){
+
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java:10]: IIRC Java creates an empty constructor automatically and you don't need to declare it yourself.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Override
+    public void execute() {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java:14]: Magic numbers in this function.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInIntakeLED.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package edu.greenblitz.tobyDetermined.commands.LED;
+
+import edu.greenblitz.tobyDetermined.OI;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class ObjectInIntakeLED extends LEDCommand{
+
+    public ObjectInIntakeLED(){
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInIntakeLED.java:9]: IIRC Java creates an empty constructor automatically and you don't need to declare it yourself.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInIntakeLED.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+    }
+
+    @Override
+    public void execute() {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInIntakeLED.java:14]: Magic numbers in this function.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
+package edu.greenblitz.tobyDetermined.commands.LED;
+
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+
+/**
+ * @author noam - a command to indicate to the human player what object he wants (he - the robot)
+ * */
+public class RobotWantLED extends LEDCommand{
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java:10]: Name isn't a good indicator of the class purpose.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+    private wantedObject object;
+
+    public RobotWantLED(wantedObject object){
+        this.object = object;
+    }
+
+    public enum wantedObject{
+        cube,cone;
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java:19]: Java Enums are conventionally in SCREAMING_SNAKE_CASE.

Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java:19]: Newlines between enum items.

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public void execute() {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java:23]: Magic numbers in this function.

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class LED extends GBSubsystem {
+
+    private static LED instance;
+    private AddressableLED addressableLED;
+    private AddressableLEDBuffer ledBuffer;
+    private Color defaultColor = Color.kGreen;
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java:15]: Perhaps define this color in RobotMap..java?

**Review by Stormageddon37 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java (hunk 11 above)
Comment (roygold8) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java:15]: I actually think its better inside the subsytem - why do you think it should move?

File: src/main/java/edu/greenblitz/tobyDetermined/OI.java [hunk 12]
```diff
@@ -1,10 +1,19 @@
 package edu.greenblitz.tobyDetermined;
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/OI.java:1]: Dont merge with those changes

**Review by roygold8 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/tobyDetermined/Robot.java [hunk 13]
```diff
@@ -1,30 +1,35 @@
 package edu.greenblitz.tobyDetermined;
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/Robot.java:1]: The only change needed in this file is the led instance

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+        led.setColor(Color.kBlue);
+        Timer.delay(0.4);
+        led.setColor(Color.kWhite);
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java:17]: We talked about color to white changes != blinks

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInIntakeLED.java [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+        led.setColor(Color.kGreen);
+        Timer.delay(0.4);
+        led.setColor(Color.kWhite);
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInIntakeLED.java:17]: We talked about color to white changes != blinks

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+        switch (this.object){
+            case cone:
+                led.setColor(Color.kYellow);
+                Timer.delay(0.4);
+                led.setColor(Color.kWhite);
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java:28]: same

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+                Timer.delay(0.2);
+                led.setColor(Color.kYellow);
+                Timer.delay(0.2);
+                break;
+            case cube:
+                led.setColor(Color.kMagenta);
+                Timer.delay(0.4);
+                led.setColor(Color.kWhite);
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/RobotWantLED.java:36]: same

**Review by roygold8 — CHANGES_REQUESTED** [review]

**Comment by roygold8** [comment]
Remember to check everything after all CR changes

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java (hunk 11 above)
Comment (Stormageddon37) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java:15]: It's not a big deal but I believe default values should be defined in a config file, all in one place so it's easy to find. It's a matter of convention - if someone wants to find a constant, they should probably go look In RobotMap, not the subsystem.

Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java:15]: The default color is changing in the game so it should be in the LED subsystem. I also think there should be a default default value at robot map

Comment (Stormageddon37) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java:15]: By definition, a default doesn't change. If you need it to change throughout the match, change the variable name.

File: src/main/java/edu/greenblitz/tobyDetermined/OI.java (hunk 12 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/OI.java:1]: what?

File: src/main/java/edu/greenblitz/tobyDetermined/Robot.java (hunk 13 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/Robot.java:1]: we need to set it's default command

File: src/main/java/edu/greenblitz/tobyDetermined/OI.java (hunk 12 above)
Comment (yonnbro) [reply src/main/java/edu/greenblitz/tobyDetermined/OI.java:1]: don't merge with the changes to the buttons, delete your button changes or if you want - put them on a 3rd\4th controller (other then the main and second controllers).

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * @author noam - a command to indicate to the human player what object he wants (he - the robot)
+ * */
+public class HumanPlayerObjectIndicator extends LEDCommand{
+
+    private wantedObject object;
+
+    public HumanPlayerObjectIndicator(wantedObject object){
+        this.object = object;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java:16]: gameObject is more descriptive

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    public enum wantedObject{
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java:19]: wantedGameObject?

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+        CUBE,
+        CONE;
+    }
+
+    @Override
+    public void initialize() {
+        switch (this.object){
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java:26]: in initialize?

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+
+
+public class ObjectInClawLED extends LEDCommand{
+
+    @Override
+    public void initialize() {
+        led.setBackgroundColor(Color.kBlue);
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java:12]: again why in initialize? move to execute?

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInIntakeLED.java [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class ObjectInIntakeLED extends LEDCommand{
+
+
+    @Override
+    public void initialize() {
+        led.setBackgroundColor(Color.kRed);
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInIntakeLED.java:13]: initialize

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java [hunk 23]
```diff
@@ -19,23 +19,7 @@ public static Dashboard init() {
 	
 	@Override
 	public void periodic() {
-		SmartDashboard.putNumber("pigeon angle", SwerveChassis.getInstance().getPigeonGyro().getYaw());
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java:22]: are you sure?

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java [hunk 24]
```diff
@@ -0,0 +1,78 @@
...
+    public void turnOff (int index){
+        setColor(index,new Color(0,0,0));
+    }
+    public void turnoff (int startIndex,int endIndex){
+        for (int i = startIndex; i < endIndex; i++) {
+            setColor(i,new Color(0,0,0));
+        }
+    }
+
+    public void setColor(int i, Color color) {
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java:53]: put the setColor functions togther!

**Review by yonnbro — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java (hunk 20 above)
Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java:26]: the whole led commands are in the init

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java [hunk 25]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * @author noam - a command to indicate to the human player what object he wants (he - the robot)
+ * */
+public class HumanPlayerObjectIndicator extends LEDCommand{
+
+    private wantedObject wantedGmaeObject;
+
+    public HumanPlayerObjectIndicator(wantedObject object){
+        this.wantedGmaeObject = object;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/LED/HumanPlayerObjectIndicator.java:16]: Game*

File: src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java (hunk 21 above)
Comment (yonnbro) [reply src/main/java/edu/greenblitz/tobyDetermined/commands/LED/ObjectInClawLED.java:12]: why resolved? if you agree lets fix if not lets talk about it

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java (hunk 23 above)
Comment (yonnbro) [reply src/main/java/edu/greenblitz/tobyDetermined/subsystems/Dashboard.java:22]: why resolve? lets not delete the dashboard values if they are unrelated to the code change, unless you absolutely know for a fact we don't and won't need them

**Review by yonnbro — CHANGES_REQUESTED** [review]

**Pull request merged**